stetho-repl-js/src/main/java/com/facebook/stetho/inspector/console/js/JavaScriptRuntimeBuilder.java

```diff
@@ -0,0 +1,248 @@
+   * @return the builder
+   */
+  public JavaScriptRuntimeBuilder importClass(Class<?> aClass) {
+    mClasses.add(aClass);
+    return this;
+  }
+
+  /**
+   * Request that the given package name will be imported in the javascript runtime.
+   * This means that all classes (enums and interfaces) will be imported.
+   * @param packageName the java package name to import
+   * @return the builder
+   */
+  public JavaScriptRuntimeBuilder importPackage(String packageName) {
+    mPackages.add(packageName);
+    return this;
+  }
+
+  /**
+   * Add a variable (binding) to the javascript environment.
+   * @param name the javascript variable name
+   * @param value the value to add
+   * @return the builder
+   */
+  public JavaScriptRuntimeBuilder addVariable(String name, Object value) {
+    variables.put(name, value);
+    return this;
+  }
+
+  /**
+   * Adds a function to the javascript environment.
+   * @param name the javascript function name
+   * @param function the function
+   * @return the builder
+   */
+  public JavaScriptRuntimeBuilder addFunction(String name, Function function) {
+    functions.put(name, function);
+    return this;
+  }
+
+  /**
+   * Initializes a proper javascript scope (runtime environment holding variables).
+   * @return a javascript scope
+   */
+  private ScriptableObject initJsScope() {
+    final Context jsContext = JavaScriptRuntimeRepl.enterJsContext();
+    try {
+      ScriptableObject scope = initJsScope(jsContext);
+      return scope;
+    } finally {
+      Context.exit();
+    }
+  }
+
+  private ScriptableObject initJsScope(Context jsContext) {
+    LogUtil.i("js) init JS scope");
+
+    // Set the main Rhino goodies
+    ImporterTopLevel importerTopLevel = new ImporterTopLevel(jsContext);
+    ScriptableObject scope = jsContext.initStandardObjects(importerTopLevel, false);
+
+    ScriptableObject.putProperty(scope, "context", Context.javaToJS(mContext, scope));
+
+    // Import the classes that the caller requested
+    for (Class<?> aClass : mClasses) {
+      String className = aClass.getName();
+      try {
+        String expression = String.format("importClass(%s)", className);
+        LogUtil.i("js) init import class: %s", expression);
+        jsContext.evaluateString(scope, expression, "chrome", 1, null);
+      } catch (Exception e) {
+        LogUtil.e(e, "js) Failed to import class: %s", className);
+        String message = String.format("Failed to import class: %s; %s", className, Log.getStackTraceString(e));
+        CLog.writeToConsole(Console.MessageLevel.ERROR, Console.MessageSource.JAVASCRIPT, message);
+      }
+    }
+
+    // Import the packages that the caller requested
+    for (String packageName : mPackages) {
+      try {
+        String expression = String.format("importPackage(%s)", packageName);
+        LogUtil.i("js) init import package: %s", expression);
+        jsContext.evaluateString(scope, expression, "chrome", 1, null);
+      } catch (Exception e) {
```

Comment:
Prefer to tidy up the exception handling with a simple `jsCLog` method that takes an exception and some error string.  Also note that you can extend `writeToConsole` to actually write the call stack to the JavaScript console (I _think_ we have that code in `Page` or `Runtime` today but I can't remember.  In any case, check out https://code.google.com/p/chromium/codesearch#chromium/src/third_party/WebKit/Source/devtools/protocol.json&q=protocol.json&sq=package:chromium&type=cs (see `Runtime.evaluate`)